mdns.h

```diff
@@ -1572,7 +1572,7 @@ mdns_record_parse_txt(const void* buffer, size_t size, size_t offset, size_t len
 		++strdata;
 		offset += sublength + 1;
 
-		size_t separator = 0;
+		size_t separator = sublength;
```

Comment:
Problem here is that a TXT record with non-ASCII data will be parsed in violation of the standard. Keep this assignment but add an assign to reset separator to 0 if non-ASCII is encountered, as well as restoring the zero check.